### PR TITLE
[Merged by Bors] - Unify common dependencies

### PR DESCRIPTION
**Pull request by c410-f3r**
There is a considerable amount of dependencies and I think it is possible to cut some transient slack in order to reduce compilation time.

Starting with the unification of common dependencies that aren't part of workspace declarations. `cargo test --all-features` didn't fail so everything should probably be OK.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -80,7 +80,6 @@ color-eyre = { version = "0.6.0", default-features = false }
 colored = "2.0.0"
 crc32c = { version = "0.6"}
 comfy-table = { version = "7.0.1", default-features = false }
-convert_case = "0.6.0"
```
Comment (c410-f3r) [inline Cargo.toml:83]: Not used

**Comment by EstebanBorai** [comment]
Hi @c410-f3r! I think we have to bump patch on `fluvio-protocol-derive`, `fluvio-socket` and `fluvio-smartmodule-derive`

**Comment by c410-f3r** [comment]
Thank you @EstebanBorai

File: connector/cargo_template/Cargo.toml [hunk 2]
```diff
@@ -7,4 +7,4 @@ edition = "2021"
 [dependencies]
 fluvio = { git = "https://github.com/infinyon/fluvio", rev = "{{fluvio-cargo-dependency-hash}}" }
 fluvio-connector-common = { git = "https://github.com/infinyon/fluvio", rev = "{{fluvio-cargo-dependency-hash}}", features = ["derive"]}
-serde = { version = "1.0", default-features = false, features = ["derive"]}
+serde = { default-features = false, features = ["derive"], workspace = true }
```
Comment (sehz) [inline connector/cargo_template/Cargo.toml:10]: templates can't have workspace dependency since they will be deployed outside fluvio repo

**Review by sehz — CHANGES_REQUESTED** [review]
Thanks for your contribution.   minor issue

File: connector/cargo_template/Cargo.toml (hunk 2 above)
Comment (c410-f3r) [reply connector/cargo_template/Cargo.toml]: Oopss... Fixed!

**Review by sehz — APPROVED** [review]

**Comment by EstebanBorai** [comment]
Hi @c410-f3r! Thanks for the time and effort invested in this PR!
Could you give an eye to the Git conflict on `crates/fluvio-smartmodule/Cargo.toml` so we can merge this?

**Comment by c410-f3r** [comment]
> Hi @c410-f3r! Thanks for the time and effort invested in this PR! Could you give an eye to the Git conflict on `crates/fluvio-smartmodule/Cargo.toml` so we can merge this?

Oh, thank you for the heads up! The conflicting file has been fixed.

**Review by EstebanBorai — APPROVED** [review]
LGTM!

**Comment by EstebanBorai** [comment]
> > Hi @c410-f3r! Thanks for the time and effort invested in this PR! Could you give an eye to the Git conflict on `crates/fluvio-smartmodule/Cargo.toml` so we can merge this?
> 
> Oh, thank you for the heads up! The conflicting file has been fixed.

Thanks to you!!

**Comment by c410-f3r** [comment]
Looks like `fluvio-smartmodule` also needs an update

**Pull request closed without merging**